mne/io/brainvision/tests/test_brainvision.py

```diff
@@ -452,4 +452,14 @@ def test_events():
     assert_equal(raw.info['chs'][-1]['ch_name'], 'STI 014')
 
 
+def test_brainvision_with_montage():
+    """Test reading embedded montage information"""
+    raw = read_raw_brainvision(vhdr_v2_path, eog=eog, misc=['ReRef'])
+    for i, d in enumerate(raw.info['dig'], 1):
+        assert_equal(d['coord_frame'], FIFF.FIFFV_COORD_HEAD)
+        assert_equal(d['ident'], i)
+        assert_equal(d['kind'], FIFF.FIFFV_POINT_EEG)
+        assert_equal(len(d['r']), 3)
```

Comment:
it does not affect the chs pos field? if so please test this

Reply:
Okay, something like this? testing that (1) if a channel is labelled eeg in one case, then it should also be eeg in the other, (2) testing equality of `loc` field for the first five eeg channels, and (3) and testing equality of `loc` field of any channels labelled eog or misc.

    raw_none = read_raw_brainvision(vhdr_v2_path)
    i, N = 0, 5
    for r,n in zip(raw.info['chs'], raw_none.info['chs']):
        if r['kind'] == FIFF.FIFFV_EEG_CH or n['kind'] == FIFF.FIFFV_EEG_CH:
            assert_equal(r['kind'], FIFF.FIFFV_EEG_CH)
            assert_equal(n['kind'], FIFF.FIFFV_EEG_CH)
            if i < N:
                assert_array_equal(r['loc'], n['loc'])
                i += 1      
        elif FIFF.FIFFV_EOG_CH in (r['kind'], n['kind']) or \
             FIFF.FIFFV_MISC_CH in (r['kind'], n['kind']):
            assert_array_equal(r['loc'], n['loc'])

Reply:
The logic looks sound to me (although the implementation isn't too Pythonesque :) ) .

Reply:
@jona-sassenhagen is seeing the sound of logic, I think that is some heavy synesthesia. 😉

Reply:
yeah it's not pretty we can agree on that. Suggestions are welcome :)

to me the most straight forward thing would probably be either (1) testing that all loc info across the two instances are equal (irrespective of type) or (2) only test the loc info of the channels whose type is different across the two instances, the latter of which could be done by for example

    for r,n in zip(raw.info['chs'], raw_none.info['chs']):
        try:
            assert_equal(r['kind'], n['kind'])
        except AssertionError:
            assert_array_equal(r['loc'], n['loc'])